PoolStageSimulator.Core/CompetitionResult.cs

```diff
@@ -2,6 +2,9 @@ namespace PoolStageSimulator.Core
 {
     public class CompetitionResult
     {
+        public Team ParticipatingTeam { get; set; }
```

Comment:
Its really hard, but in a TDD process, this code should have been added after the test failed.
If I can remember correctly, there are 3 stages of TDD which are:
- Red: Test is written, but code so it doesn't compile (you can think this as the compiler that is also a test tool)
- Yellow (code compiles, but tests fails)
- Green (add as little code to pass tests)

Also note, if all tests pass, but there are still bugs, you still write the missing tests first. Never change production code except to make tests pass.

TDD is explained using kotlin in the following [video](https://www.youtube.com/watch?v=qkblc5WRn-U)

Reply:
This might be a minor issue, but I still brought this up since the commit doesn't exactly say what it does.

Reply:
The problem isn't the TDD process in this case, but the commit which includes more than what it does. 

I can make use of cherry picking to change this if necessary, but I don't think this is such a big issue. The code was changed since the tests made the compiler to fail, so there is a guarantee the code was changed after the test.